docs/transpile/custom-transpiler-pass.ipynb

```diff
@@ -0,0 +1,239 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Transpiler Passes\n",
+    "\n",
+    "Transpiler passes are classified into one of two classes: Analysis passes and Transformation passes. Passes in general work with the Directed Acyclic Graph (DAG) **TODO: insert link here** and the `property_set`, a dictionary-like object for storing properties determined by analysis passes. Analysis passes work with the DAG, but are not allowed to modify it. This contrasts with transformation passes, which do modify the DAG, and can read (but not write to) `property_set`. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Internals\n",
+    "\n",
+    "**TODO: put something here, but I do not know what**"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Pauli Twirling\n",
+    "\n",
+    "As an example we construct a transpiler pass that adds Pauli twirls. Pauli twirling is an error suppression strategy that randomizes how the qubits experience noisy channels, which we assume to be two-qubit gates below (because they are much more error-prone than single-qubit gates). However, the Pauli twirls do not effect the actual operation of the two-qubit gate, they are chosen such that those applied *before* the two-qubit gate (to the left) are countered by those applied *after* the two-qubit gate (to the right). In this sense, the two-qubit operations are identical but the way they are performed is different. One benefit of Pauli twirling is that it turns coherent errors into stochastic errors, which can be improved by averaging more.\n",
```

Comment:
```suggestion
    "As an example we construct a transpiler pass that adds Pauli twirls. Pauli twirling is an error suppression strategy that randomizes how the qubits experience noisy channels, which we assume to be two-qubit gates in this example (because they are much more error-prone than single-qubit gates). However, the Pauli twirls do not affect the actual operation of the two-qubit gate. They are chosen such that those applied *before* the two-qubit gate (to the left) are countered by those applied *after* the two-qubit gate (to the right). In this sense, the two-qubit operations are identical but the way they are performed is different. One benefit of Pauli twirling is that it turns coherent errors into stochastic errors, which can be improved by averaging more.\n",
```

Reply:
" One benefit of Pauli twirling is that it turns coherent errors into stochastic errors, which can be improved by averaging more."  The "by averaging more" at the end is unclear to me.  Are you saying that stochastic errors can be improved by averaging (averaging what?), or...